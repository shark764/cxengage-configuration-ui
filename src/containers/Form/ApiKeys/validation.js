/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

import { isEmpty } from 'serenova-js-utils/strings';

export const formValidation = values => ({
  name: isEmpty(values.get('name')) && 'Please enter a name',
  roleId: !values.get('roleId') && 'You must select a role'
});
