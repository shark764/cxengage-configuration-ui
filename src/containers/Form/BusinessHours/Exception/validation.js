export const formValidation = values => ({
  startTimeMinutes:
    values.get('startTimeMinutes') > values.get('endTimeMinutes') &&
    values.get('endTimeMinutes') !== 0 &&
    'Start time should be ahead of end time.',
  date:
    values.get('date') &&
    new Date(
      values
        .get('date')
        .toJSON()
        .split('T')[0]
    ) < new Date(new Date().toJSON().split('T')[0]) &&
    'Please select a future date'
});
