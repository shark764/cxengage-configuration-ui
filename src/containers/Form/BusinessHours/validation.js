/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

import { isEmpty } from 'serenova-js-utils/strings';

export const formValidation = values => {
  const daysInitials = ['sun', 'sat', 'mon', 'tue', 'thu', 'wed', 'fri'];
  let validation = {};

  validation.name = isEmpty(values.get('name')) && 'Please enter a name';
  return daysInitials.reduce(
    (r, i) => ({
      ...r,
      [i.concat('StartTimeMinutes')]:
        values.get(i.concat('StartTimeMinutes')) > values.get(i.concat('EndTimeMinutes')) &&
        values.get(i.concat('EndTimeMinutes')) !== 0 &&
        'Start time may not be after end time'
    }),
    validation
  );
};
