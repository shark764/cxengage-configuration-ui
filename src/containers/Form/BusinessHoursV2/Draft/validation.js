import { isEmpty } from 'serenova-js-utils/strings';

export const formValidation = values => ({
  name: isEmpty(values.get('name')) && 'Please enter a name'
});