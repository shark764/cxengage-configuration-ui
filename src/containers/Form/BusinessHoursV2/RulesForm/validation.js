import { isEmpty } from 'serenova-js-utils/strings';

const repeats = {
  weekly: 'every how many weeks',
  daily: 'every how many days',
  monthly: 'every how many months'
};

const daysOnMonth = {
  january: 31,
  febraury: 29, //Max number fo days febraury can have (leap year)
  march: 31,
  april: 30,
  may: 31,
  june: 30,
  july: 31,
  august: 31,
  september: 30,
  october: 31,
  november: 30,
  december: 31
};

export const formValidation = values => {
  let errors = {};
  let rulesErrors = [];
  if (values.get('rules') && values.get('rules').size) {
    values.get('rules').forEach((rule, index, rules) => {
      const error = ruleValidation(rule, rules);
      if (error) {
        rulesErrors[index] = error;
      }
    });
  }

  if (rulesErrors.length) {
    errors.rules = rulesErrors;
  }
  return errors;
};

export const ruleValidation = (rule, rules) => {
  let error;
  if (isEmpty(rule.name)) {
    error = {
      name: true,
      message: 'The rule should have a name'
    };
  } else if (rules.some(r => r.name.trim() === rule.name.trim() && !isEmpty(rule.name) && r.id !== rule.id)) {
    error = {
      name: true,
      message: 'Rule name should be unique within the draft'
    };
  } else if (isEmpty(rule.type)) {
    error = {
      type: true,
      message: 'The rule should have a type'
    };
  } else if (!rule.startDate) {
    error = {
      startDate: true,
      message: 'You should select a start date'
    };
  } else if (rule.startDate && !rule.type.includes('one-time') && rule.endDate && rule.startDate > rule.endtDate) {
    error = {
      endDate: true,
      message: 'end date should be after start date'
    };
  } else if (!rule.type.includes('one-time') && !rule.repeats) {
    error = {
      repeats: true,
      message: 'You should select the rate at which the rule repeats'
    };
  } else if (!rule.type.includes('one-time') && !rule.every) {
    error = {
      every: true,
      message: `You should set ${repeats[rule.repeats]} should the rule take effect`
    };
  } else if (!rule.type.includes('one-time') && isNaN(rule.every) && rule.repeats !== 'yearly') {
    error = {
      every: true,
      message: 'The Every field should be a number'
    };
  } else if (!rule.type.includes('one-time') && !Number.isInteger(+rule.every) && rule.repeats !== 'yearly') {
    error = {
      every: true,
      message: 'The Every field should be a integer value'
    };
  } 
  else if (!rule.type.includes('one-time') && rule.repeats === 'weekly' && (!rule.on || !Array.isArray(rule.on) || rule.on.length < 1)) {
    error = {
      on: true,
      message: 'You should select the days on the week in which the rule should take effect'
    };
  } else if (!rule.type.includes('one-time') &&
    ((rule.repeats === 'monthly' || rule.repeats === 'yearly') && (!rule.on || !rule.on.type || !rule.on.value)))  {
    error = {
      on: {
        type: !rule.on || !rule.on.type,
        value: !rule.on || !rule.on.value
      },
      message: 'type or value on the "On" field is undefined'
    };
  } else if (!rule.type.includes('one-time') &&
  ((rule.repeats === 'monthly' || rule.repeats === 'yearly') && 
  (rule.on &&
    rule.on.value === 'day' &&
    (rule.on && (isNaN(rule.on.type) || !Number.isInteger(+rule.on.type)))))) {
      error = {
        on: {
          type: true
        },
        message: 'value on the "On" field should be a number'
      };
  } else if (
    !rule.type.includes('one-time') &&
    rule.repeats === 'monthly' &&
    rule.on &&
    rule.on.value === 'day' &&
    (rule.on.type < 1 || rule.on.type > 31)
  ) {
    error = {
      on: {
        type: true,
      },
      message: 'The numeric value in the "On" field has to be between 1 and 31(days on a month)'
    };
  } else if (
    !rule.type.includes('one-time') &&
    rule.repeats === 'yearly' &&
    rule.on &&
    rule.on.value === 'day' &&
    rule.every &&
    (rule.on.type < 1 || rule.on.type > daysOnMonth[rule.every])
  ) {
    error = {
      on: {
        type: true
      },
      errorMessage: `The numeric value in the "On" field has to be between 1 and ${
        daysOnMonth[rule.every]
      } (Number of days on ${rule.every}, considering a day more when leap year on Febraury)`
    };
  } else if (
    rule.hours &&
    !rule.hours.allDay &&
    rule.hours.intervals &&
    rule.hours.intervals.findIndex(
      (interval, ind, intervals) =>
        interval.start > interval.end ||
        intervals.some(
          (int, i) =>
            i !== ind &&
            ((interval.start >= int.start && interval.start <= int.end) ||
              (interval.end >= int.start && interval.end <= int.end))
        )
    ) > -1
  ) {
    const index = rule.hours.intervals.findIndex(
      (interval, ind, intervals) =>
        interval.start > interval.end ||
        intervals.some((int, i) => i !== ind && ((interval.start >= int.start && interval.start <= int.end) || 
          (interval.end >= int.start && interval.end <= int.end)))
    );
    let intervalErrors = Array(rule.hours.intervals.length).fill({
      start: false,
      end: false
    });
    intervalErrors[index] = {
      start: true,
      end: true
    };

    error = {
      hours: {
        intervals: intervalErrors
      },
      message: 'An interval has a greater start time than its end time'
    };
  }

  return error;
}
