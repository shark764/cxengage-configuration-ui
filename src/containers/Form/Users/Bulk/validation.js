/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

export const formValidation = values => ({
  status: !values.get('status') && 'Please select a status',
  noPassword: !values.get('noPassword') && 'Please select a value for Platform Authentication',
  defaultIdentityProvider:
    !values.get('defaultIdentityProvider') && 'Please select a value for Default Identity Provider'
});
