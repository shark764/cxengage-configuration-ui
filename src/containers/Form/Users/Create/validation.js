/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { isEmpty } from 'serenova-js-utils/strings';
import { validateEmail } from 'serenova-js-utils/validation';

const emailValidation = email => {
  if (isEmpty(email)) {
    return 'Please enter an email';
  } else if (!validateEmail(email)) {
    return 'Please enter a valid email address';
  } else {
    return false;
  }
};

export const formValidation = (values, props) => {
  const allUsers = props.tenantUsers.toJS();
  const userIndex = allUsers.findIndex(element => element.email === values.get('email'));
  const platformRoleEnabled = !props.checkPlatformUser;

  if (userIndex !== -1) {
    props.setSelectedEntityId(allUsers[userIndex].id);
  }
  return {
    roleId: !values.get('roleId') && 'Please select a Tenant Role',
    email: emailValidation(values.get('email')),
    platformRoleId: platformRoleEnabled && !values.get('platformRoleId') && 'Please select a Platform Role'
  };
};
