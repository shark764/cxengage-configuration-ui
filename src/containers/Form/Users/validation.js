/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */
import { validatePhoneNumber, validateSip } from 'serenova-js-utils/validation';
import { isEmpty } from 'serenova-js-utils/strings';

export const formValidation = values => {
  const formValidation = {};
  const newValues = values.toJS();
  const bypassNamesValidation =
    isEmpty(values.get('firstName')) &&
    isEmpty(values.get('lastName')) &&
    (values.get('invitationStatus') === 'pending' ||
      values.get('invitationStatus') === 'invited' ||
      values.get('invitationStatus') === 'expired');

  const extensionsErrors = newValues.extensions.map(({ description, type, value }) => ({
    ...(isEmpty(description) && {
      label: true,
      message: 'Please provide a description'
    }),
    ...(((type === 'pstn' && !validatePhoneNumber(value)) ||
      ((type === 'sip' && !validateSip(value)) || (type !== 'webrtc' && !value))) && {
      value: true,
      message:
        type !== 'webrtc' && !value
          ? 'Value required'
          : type === 'sip' ? "Extensions must start with 'sip:'." : 'Phone number should be in E.164 format.'
    })
  }));

  if (extensionsErrors.some(({ label, value }) => label || value)) {
    formValidation.extensions = extensionsErrors;
  }

  formValidation.firstName = !bypassNamesValidation && isEmpty(values.get('firstName')) && 'Please enter a first name';
  formValidation.lastName = !bypassNamesValidation && isEmpty(values.get('lastName')) && 'Please enter a last name';
  formValidation.roleId = !values.get('roleId') && 'Please select a Tenant Role';
  return formValidation;
};
