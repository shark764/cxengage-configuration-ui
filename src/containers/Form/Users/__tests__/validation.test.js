import { formValidation } from '../validation';
import { Map } from 'immutable';

describe('formValidation', () => {
  it('returns proper object when required fields are provided', () => {
    const values = new Map({
      email: 'mockEmail',
      platformRoleId: 'mockPlatformRoleId',
      roleId: 'mockRoleId',
      firstName: 'mockFirstName',
      lastName: 'mockLastName',
      externalId: '1234',
      workStationId: '1234',
      extensions: [
        { type: 'pstn', value: '15067994185' },
        { type: 'sip', value: 'sip:01633973283@ns.subdomain.company.com' }
      ]
    });
    expect(formValidation(values)).toMatchSnapshot();
  });
  it('returns proper object (errors) when required fields are not provided', () => {
    const values = new Map({
      email: '',
      platformRoleId: '',
      roleId: '',
      firstName: 'mockFirstName',
      lastName: 'mockLastName',
      externalId: '1234',
      workStationId: '1234',
      extensions: [
        { type: 'pstn', value: '15067994185' },
        { type: 'sip', value: 'sip:01633973283@ns.subdomain.company.com' }
      ]
    });
    expect(formValidation(values)).toMatchSnapshot();
  });
  it('returns proper object (errors) when wrong values are provided', () => {
    const values = new Map({
      email: undefined,
      platformRoleId: null,
      roleId: null,
      firstName: undefined,
      lastName: undefined,
      externalId: undefined,
      workStationId: undefined,
      extensions: [{ type: 'pstn', value: 'rafggfdg' }, { type: 'sip', value: 'asdasd' }]
    });
    expect(formValidation(values)).toMatchSnapshot();
  });
});
