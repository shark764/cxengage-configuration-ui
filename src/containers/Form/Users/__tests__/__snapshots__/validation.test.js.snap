// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formValidation returns proper object (errors) when invitationStatus is pending, firstname is provided so lastname should be required as well 1`] = `
Object {
  "extensions": Array [
    Object {
      "label": true,
      "message": "Phone number should be in E.164 format.",
      "value": true,
    },
    Object {
      "label": true,
      "message": "Please provide a description",
    },
  ],
  "firstName": false,
  "lastName": "Please enter a last name",
  "roleId": false,
}
`;

exports[`formValidation returns proper object (errors) when required fields are not provided 1`] = `
Object {
  "extensions": Array [
    Object {
      "label": true,
      "message": "Phone number should be in E.164 format.",
      "value": true,
    },
    Object {
      "label": true,
      "message": "Please provide a description",
    },
  ],
  "firstName": false,
  "lastName": false,
  "roleId": "Please select a Tenant Role",
}
`;

exports[`formValidation returns proper object (errors) when wrong values are provided 1`] = `
Object {
  "extensions": Array [
    Object {
      "label": true,
      "message": "Phone number should be in E.164 format.",
      "value": true,
    },
    Object {
      "label": true,
      "message": "Extensions must start with 'sip:'.",
      "value": true,
    },
  ],
  "firstName": "Please enter a first name",
  "lastName": "Please enter a last name",
  "roleId": "Please select a Tenant Role",
}
`;

exports[`formValidation returns proper object when required fields are provided 1`] = `
Object {
  "extensions": Array [
    Object {
      "label": true,
      "message": "Phone number should be in E.164 format.",
      "value": true,
    },
    Object {
      "label": true,
      "message": "Please provide a description",
    },
  ],
  "firstName": false,
  "lastName": false,
  "roleId": false,
}
`;

exports[`formValidation returns proper object when required fields are provided, invitationStatus is pending so the firstname & lastname are not required 1`] = `
Object {
  "extensions": Array [
    Object {
      "label": true,
      "message": "Phone number should be in E.164 format.",
      "value": true,
    },
    Object {
      "label": true,
      "message": "Please provide a description",
    },
  ],
  "firstName": false,
  "lastName": false,
  "roleId": false,
}
`;
